### DISCUSSION TITLE
Date format on the graphs

### DISCUSSION THREAD

**Discussion by Ecloud44** [Daveiano/weewx-wdc#290] [Q&A]
Hi
How to change the date format on the graphs
The date is good on the header of the page, but not on the graph

have a look ont ​​the screen shot

![homepageWdcFrDate](https://github.com/user-attachments/assets/94185017-86a0-43fd-b1c9-6f3f337efa62)

![GraphWdcEnDate](https://github.com/user-attachments/assets/eb86b5a7-6135-4d5b-ae39-8382c9e08abd)

https://meteo.inodes.fr/wdc/index.html

Thank you for your help
Remy

**Comment by Daveiano** [comment]
Hi! 

There is a section about this in the wiki: https://github.com/Daveiano/weewx-wdc/wiki/Configuration#configuring-date-and-time-formats

Please have a look, it should answer your question. Feel free to ask if you have any questions left!

**Comment by Ecloud44** [comment]
Hi 
i 'am afraid but i don't see where i can change date format .
In my weewx.conf is setup to local format .
like :lun. 23 déc. 2024 11:36:52 CET 


 [[WdcReport]]
        HTML_ROOT = /var/www/html/Wdc
        lan = fr
        skin = weewx-wdc
        enable = true

I use Bootstap skin to compare , with te same parameters
see screen shot 

![comapre-wdc-bst](https://github.com/user-attachments/assets/50780646-724b-4126-8fb7-3a2ca4076ba5)

Another question , i d'ont have the same resultat it is because  with wdc i plot Gust speed average ? Can I change it ?
Regards , 
Remy

**Comment by Daveiano** [comment]
> i 'am afraid but i don't see where i can change date format .

Please have a look at skin.conf at `[DisplayOptions][diagrams][day]` for example:

```
[[[day]]]
            aggregate_interval = 1800 # 30 minutes
            bottom_date_time_format = "%H:%M"
            tooltip_date_time_format = "%x %X"
            [[[[observations]]]]
                [[[[[tempdew]]]]]
                [[[[[outHumidity]]]]]
                [[[[[barometer]]]]]
                [[[[[windchill_heatindex]]]]]
                [[[[[wind]]]]]
                [[[[[windDir]]]]]
                [[[[[windRose]]]]]
                [[[[[rain]]]]]
                    aggregate_interval = 7200  # 2 hours
                    tooltip_date_time_format = "%X"
                [[[[[rainRate]]]]]
                [[[[[UV]]]]]
                [[[[[ET]]]]]
                    aggregate_interval = 7200  # 2 hours
                    tooltip_date_time_format = "%X"
                [[[[[radiation]]]]]
                [[[[[cloudbase]]]]]
                [[[[[appTemp]]]]]
```

For each context (day, week, month) there are predefined date formats via `bottom_date_time_format` and `tooltip_date_time_format`. Here is a help page for getting the right format: https://d3-wiki.readthedocs.io/zh-cn/master/Time-Formatting/

You can set a date format for the whole context (eg. day) or individually for each chart if you want to.

> Another question , i d'ont have the same resultat it is because with wdc i plot Gust speed average ? Can I change it ?

Sure! You just need to set the right `aggregate_type`, eg:

```
    [[[[wind]]]]
                label = 'Wind speed / Gust speed'
                yScaleMin = 0
                enableArea = True
                areaOpacity = 0.5
                curve = linear
                [[[[[obs]]]]]
                    [[[[[[gust]]]]]]
                        observation = "windGust"
                        color = "#666666"
                        aggregate_type = "max"
                    [[[[[[speed]]]]]]
                        observation = "windSpeed"
                        color = "#ffc000"
                        aggregate_type = "max"
```

**Comment by Ecloud44** [comment]
Hi
Thank a lot for your response , it  work now with the modiifcation.
Regards
